manifests/cf-manifest/manifest/020-cf-properties.yml

```diff
@@ -666,6 +666,9 @@ properties:
       preloaded_rootfses:
         - "cflinuxfs2:/var/vcap/packages/cflinuxfs2/rootfs"
 
+    executor:
+      memory_capacity_mb: (( calc "floor(30662 * 1.8)" ))
```

Comment:
Can you (or anyone else) think of a way to make this more robust against us changing the instance type in the future? The old Warden/DEA implementation had an `memory_overcommit_factor` option, but I can't find any equivalent feature or discussion of such a feature for Diego.

https://github.com/alphagov/paas-cf/blob/cd251bdc50994a3bf2f2451db81775e57a3b4c1b/manifests/cf-manifest/cloud-config/020-cf-vm-types.yml#L62

Something that we might be able to change at the same time is that the `30662` number feels a bit magic and like it would benefit from using `(( grab DESCRIPTIVE_NAME ))`.

Reply:
You're absolutely right, Dan. `30662` *is* magic. So I'm looking into it.

* From [the use of `1024 * 1024` for `MB`](https://github.com/cloudfoundry/executor/blob/master/initializer/configuration/configuration.go#L49-L59 ), `executor` is using `MiB`.
* `31.5GiB` is `31232MiB`. This is `570MiB` more than what `rep` was advertising.
* `30662` matches the total memory reported by `free -m` on a dev env `cell/0`:
  ```
  free -mcell/7c767cf5-30c0-49d6-8711-803c2fbe425e:~$ free -m
               total       used       free     shared    buffers     cached
  Mem:         30662       9231      21430          6        241       6204
  -/+ buffers/cache:       2786      27875
  Swap:        30663          0      30663
  ```
* Following `garden.Capacity.MemoryInBytes` a long way down it eventually gets the VM's total memory capacity.
* So, `30662` is the VM's total memory capacity. @dcarley any idea if that `570MB` is special?

Reply:
I think it's important to give a name to the number so that someone reading it understands what it represents. I'm not so concerned about the precision of the actual number - saying in the commit message where we got it from seems good enough.